src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -97,6 +107,7 @@
     private NotificationCompat.Builder mNotificationBuilder;
     private int mLastPercent;
 
+    private static int jobId = 0;
```

Comment:
fileId is unique; the problem is that uploads of new files have no fileId, it's set when the upload is succeeded . A big legacy mistake.